src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -80,16 +97,63 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
         }
         catch (Exception ex)
         {
-            Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
-            await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace, cancellationToken);
+            if (ex is JsonException or CmsEventException)
+            {
+                Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
+                await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace,
+                    cancellationToken);
+            }
+
+            var deliveryAttempts = 0;
+
+            if (message.ApplicationProperties.TryGetValue(CustomMessageProperty, out object? attemptObj) &&
+                int.TryParse(attemptObj?.ToString(), out int existingAttempt))
+            {
+                deliveryAttempts = existingAttempt;
+            }
+
+            if (deliveryAttempts >= _maxMessageDeliveryAttempts)
+            {
+                Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
```

Comment:
Is it worth amending the message to specifically state that it's reached the maximum redelivery attempt?